### PR TITLE
feat(benchmark): restore text model qualification

### PR DESCRIPTION
**Pull request by chaofengw-nv**
## Background

PR #1093 removed the centralized Accuracy and Performance registrations during the family-isolation cutover. PR #1204 restored the family-local qualification foundation. This follow-up restores the pre-refactor benchmark coverage without reintroducing a central model registry or binding model profiles to CI hardware.

## Exit Criteria

- Auto-discover Accuracy and Performance cases only from each owning family's `tests/benchmark/` folder.
- Restore the pre-#1093 checkpoints, precisions, prompts, generation lengths, datasets, metrics, and comparison contracts for the supported text, translation, and encoder workloads.
- Compare converted TRTMC output against Hugging Face output for Accuracy.
- Prefer `torch.compile` for Performance reference timing and use eager only as a declared fallback.
- Allow a family-specific Python reference environment only when the common environment is incompatible.
- Produce complete JSON and HTML reports when a model misses its gate; reserve `error` for framework, configuration, environment, build, execution, or reporting failures.
- Keep BART out of this change until the shared Accuracy reference has an explicit generation-policy override contract.

## Implementation

- Added 41 family-local model profiles, each declaring its own checkpoint, model options, build inputs, cases, and gates. The catalog now contains 43 profiles and 86 cases including the two profiles already present on `main`.
- Kept shared benchmark definitions reusable across families. They own benchmark datasets, metrics, and model-agnostic execution contracts; they do not contain a central model list.
- Restored reusable contracts for five-shot MMLU continuation, HumanEval continuation, WikiText-103 continuation, translation parity, STSBenchmark embedding parity, text generation Performance, and encoder/embedding Performance.
- Added generic Hugging Face reference support for causal generation, seq2seq translation, encoders, and embeddings. Family profiles may declare fully qualified Transformers reference classes and pinned reference requirements without adding model-specific branches to the shared runner.
- Kept device selection and CI resource policy in the high-level automation layer. Model profiles do not select, require, or exclude hardware.
- Aligned embedding Performance timing with the public operation boundary: compile before warmup, include host result materialization in the timed call, and perform report validation after sampling.
- Fixed text-generation Performance references to override checkpoint beam-search defaults with `num_beams=1`, matching the candidate's greedy contract.
- Made NLLB source-language token placement an explicit family-owned profile declaration, consumed generically by the shared Accuracy and Performance runners.
- Classified output-contract mismatches as model `failed` results while preserving reference, candidate, environment, and inconclusive execution failures as `error` results with their reports.
- Preserved offline operation by pinning trusted remote code, forwarding immutable revisions, and ignoring unused TensorFlow and Flax checkpoint variants during snapshot materialization.

### Restored coverage

- Text generation: Bloom, DeepSeek-V2, Falcon, Falcon3, Gemma, GLM, GPT-Neo, Granite, InternLM2, Mamba, Minitron, Mistral, Mixtral, Nemotron, OLMo, OLMo2, OPT, Phi-3, Pythia, Qwen3, RWKV, StableLM, and XGLM variants.
- Code and language-model datasets: CodeGen and StarCoder2 on HumanEval; DistilGPT2 on WikiText-103.
- Translation: Marian, T5, NLLB-200, and Riva Translate.
- Encoder and embedding: ALBERT, BERT, MiniLM, DeBERTa, DistilBERT, DPR, ModernBERT, MPNet, and RoBERTa.

### Change categories

- [ ] Model or runtime behavior
- [ ] Public API
- [ ] ABI
- [ ] Bundle or artifact format
- [ ] Dependencies
- [ ] Documentation only
- [x] CI or developer tooling

### Commands and Results

- `python3 -m pytest -q core/builder/tests/test_build_cli.py::test_hugging_face_model_id_resolves_to_a_local_snapshot tools/tests/test_model_benchmark.py tools/tests/test_architecture.py apps/benchmark/trtmc_benchmark/tests/test_perf_matrix.py`: 297 passed on exact head `0c6ae3039`.
- `ruff check <changed Python files>` and applicable `pre-commit` whitespace, EOF, YAML, and C++ format checks: passed.
- Recompiled the current family loader, fake-family DSO, and `test_family_loader` against the current headers after the CI fixture correction on exact head `644ac0a64`: `ALL PASSED`.
- Automatic discovery reports 43 profiles and 86 cases; all 41 profiles added by this PR have both Accuracy and Performance cases.

### Hardware, Environment, and Revisions

NVIDIA GB300, CUDA 13.0, TensorRT 11.2.0.113, PyTorch 2.12.0, and Transformers 5.2.0 common environment. InternLM uses its family-pinned Transformers 4.46.3 reference environment.

- All 82 cases introduced by this PR have produced a complete `passed` or `failed` result on GB300-2; none remains unexecuted.
- Exact head `3d407610f`: all 18 encoder/embedding cases passed; the 8 translation cases completed without framework errors (6 passed and 2 retained model-gate failures).
- Exact head `5cc7e0576`: CodeGen and DistilGPT2 Accuracy/Performance passed; StarCoder2 Performance passed and its Accuracy gate failure was retained.
- Exact heads `39e401159`, `d46a37f1d`, and `e9932f131`: the restored five-shot MMLU groups completed Accuracy and Performance without framework errors. Model-gate failures were retained rather than converted to passes.
- Exact head `7680d8e59`: the 12-case review-fix regression completed with 10 passes and 2 retained model Accuracy gate failures. All 6 Performance cases used successful `torch-compile` references; no case returned `error`.
- Exact head `5c56f6d6d`: MiniLM Accuracy passed. Performance retained its model gate failure after two successful `torch-compile` measurements; configuration, execution, fallback, and report generation completed without an `error`.
- Exact head `0c6ae3039`: Marian Performance passed its exact-token contract with a successful `torch-compile` reference (5.01 ms candidate / 30.31 ms reference). NLLB Performance also passed with a successful `torch-compile` reference (13.05 ms / 87.23 ms); NLLB Accuracy completed normally and retained its existing 2/10 model-gate result as `failed`.
- A negative-path run on exact head `0c6ae3039` confirmed that a dataset checksum mismatch and a candidate command failure are reported as `error`, rather than being mislabeled as model gate failures.

### Not Run / Remaining Gaps

- BART remains deferred because its old comparison requires explicit generation-policy controls that are not yet represented by the shared Accuracy reference.
- Thor targets were not run; this PR records GB300 target evidence.

## Contributor Self-Review

- [x] I have completed a self-review of this change.

## Notes For Future Readers

- A model qualification `failed` result is benchmark evidence; it must not be changed to `passed`. Configuration, environment, build, execution, and reporting failures are separate `error` results and must be fixed.
- L0 files are excluded structurally: discovery scans only `families/<family>/tests/benchmark/*.yaml`.
- Adding a compatible model requires only a family-local profile. Shared code changes are needed only for a genuinely new dataset, metric, task contract, or model-agnostic execution capability.
- The installed user benchmark remains under `apps/benchmark`; `tools/benchmark_qualification` is the separate internal automation and reporting layer.

### Risk level

- [ ] Low
- [x] Medium
- [ ] High

The profiles are family-local. Shared changes are limited to reusable benchmark contracts, reference execution, isolated Python environments, reporting continuity, and offline checkpoint materialization; these surfaces have unit and GB300 coverage.

### CODE REVIEW

**Comment by chaofengw-nv** [comment]
Superseded by the consolidated benchmark qualification PR #1379, which contains this change. Continuing validation and fixes there.
